qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -226,6 +240,26 @@ def result(self, timeout: Optional[float] = None, wait: float = 5) -> Result:
 
         return self._result
 
+    def _retrieve_partial_result(self) -> Optional[Result]:
+        """Attempt to retrieve partial results for a job.
+
+        Returns:
+            The partial results for a job, or ``None`` if schema validation fails.
+        """
+        if not self._result and not self._partial_result:  # type: ignore[misc]
+            # No results, nor partial results, have been attained for this job yet.
+            with api_to_job_error():
+                # Set partial results, in case validation fails they are cached.
+                self._partial_result = self._api.job_result(self.job_id(), self._use_object_storage)
+
+            try:
+                return Result.from_dict(self._partial_result)
+            except ValidationError:
```

Comment:
Did you verify if it really throws a `ValidationError`? I `from_dict()` converts that to a different error.

Reply:
Thanks for that catch! It actually returns a `ModelValidationError` which is one of our custom exceptions. I'll update that.